File: Plugins/Aspose-Pdf-Java-for-Jython/asposepdf/WorkingWithDocumentObject/Optimize.py
from asposepdf import Settings
from com.aspose.pdf import Document
from com.aspose.pdf.Document import OptimizationOptions

class Optimize:

    def __init__(self):
       self.optimize_web()
       
    def optimize_web(dataDir):
        
        dataDir = Settings.dataDir + 'WorkingWithDocumentObject/Optimize/'
        
        # Open a pdf document.
        doc = Document(dataDir + "input1.pdf")

        # Optimize for web
        doc.optimize()

        #Save output document
        doc.save(dataDir + "Optimized_Web.pdf")

        print "Optimized PDF for the Web, please check output file."
 
if __name__ == '__main__':        
    Optimize()

File: plugins/Aspose_Pdf_Java_for_PHP/src/Aspose/Pdf/WorkingWithDocumentObject/Optimize.php
<?php

namespace Aspose\Pdf\WorkingWithDocumentObject;

use com\aspose\pdf\Document as Document;
use com\aspose\pdf\Document\OptimizationOptions as OptimizationOptions;

class Optimize {

    public static function run($dataDir=null)
    {
        # Optimize document for the web
        Optimize::optimize_web($dataDir);

    }

    public static function optimize_web($dataDir=null)
    {

        # Open a pdf document.
        $doc = new Document($dataDir . "input1.pdf");

        # Optimize for web
        $doc->optimize();

        #Save output document
        $doc->save($dataDir . "Optimized_Web.pdf");

        print "Optimized PDF for the Web, please check output file." . PHP_EOL;

    }

}